c/misra/src/rules/RULE-9-7/UninitializedAtomicObject.ql

```diff
@@ -29,18 +30,14 @@ class ThreadSpawningFunction extends Function {
   }
 }
 
-class AtomicInitAddressOfExpr extends FunctionCall {
-  Expr addressedExpr;
+private string atomicInit() { result = "atomic_init" }
 
+class AtomicInitAddressOfExpr extends AddressOfExpr {
   AtomicInitAddressOfExpr() {
-    exists(AddressOfExpr addrOf |
-      getArgument(0) = addrOf and
-      addrOf.getOperand() = addressedExpr and
-      getTarget().getName() = "__c11_atomic_init"
+    exists(StdFunctionOrMacro<C11FunctionWrapperMacro, atomicInit/0>::Call c |
```

Comment:
I think we should encapsulate this within the `StdFunctionOrMacro` library, and expose it like this:
```cpp
class AtomicInitCall = StdFunctionOrMacro<C11FunctionWrapperMacro, atomicInit/0>::Call
```

This enables us to hide away the macro/function complexity, and provide a uniform interface for other APIs in the future.

Reply:
Great suggestion, very much agreed, done!